### PR TITLE
Add leg skin spatial calibration

### PR DESCRIPTION
**Pull request by BoPot**
The calibrations are created from provided CAD models of upper and lower leg covers. Source code and files can be found here: [https://gitlab.fel.cvut.cz/body-schema/icub/icub-skin-calibration/-/tree/main/skin_models](https://gitlab.fel.cvut.cz/body-schema/icub/icub-skin-calibration/-/tree/main/skin_models)

_CStransformations_ and _CSlabels_ .npy files contain transformations to triangle centers extracted from CAD files and corresponding triangle IDs, respectively. _skin_from_CS.py_ uses these files and .ini files from [https://github.com/robotology/icub-main/tree/master/app/skinGui/conf/skinGui](https://github.com/robotology/icub-main/tree/master/app/skinGui/conf/skinGui) to generate taxel position files (visualization included).
_calibrations_ folder contains the position files and _read_calib.py_ script for visualization.

@matejhof @rustlluk @traversaro @maggia80 @fiorisi

### CODE REVIEW

File: app/skinGui/conf/positions/left_leg_lower.txt [hunk 1]
```diff
@@ -0,0 +1,773 @@
+name	lower left leg
```
Comment (pattacini) [inline app/skinGui/conf/positions/left_leg_lower.txt:1]: I would go here with underscore-separated names.
The same applies to all files.

**Review by pattacini — CHANGES_REQUESTED** [review]
Thanks a lot @BoPot for your contribution 🚀 

I would require you to apply a few changes before merging.

Also, a couple of questions:
- https://gitlab.fel.cvut.cz/body-schema/icub/icub-skin-calibration/-/tree/main/skin_models is not accessible. Would you consider somehow making it available? Alternatively, we could host that code in `icub-main`, somewhere.
- The files you uploaded do not seem to be referenced from within he GUI conf files. Did you plan on pushing them later on?

File: app/skinGui/conf/positions/left_leg_lower.txt (hunk 1 above)
Comment (traversaro) [reply app/skinGui/conf/positions/left_leg_lower.txt:1]: > I would go here with underscore-separated names. The same applies to all files.

For all other files we used the name of the corresponding link (see https://icub-tech-iit.github.io/documentation/icub_kinematics/icub-model-naming-conventions/icub-model-naming-conventions/#links), so to keep consistency I would go for:
* `l_lower_leg`
* `l_upper_leg`
* `r_lower_leg`
* `r_upper_leg`

Comment (BoPot) [reply app/skinGui/conf/positions/left_leg_lower.txt]: I have edited the names in the headers according to the convention.
Sorry about the link issues in the comment, I have fixed that, too, they should work now.
I was not aware of the references in GUI conf files, I will try to add them where needed.

**Comment by pattacini** [comment]
>I have edited the names in the headers according to the convention.
Sorry about the link issues in the comment, I have fixed that, too, they should work now.

Thanks a lot 👍🏻 

>I was not aware of the references in GUI conf files, I will try to add them where needed.

I meant those files in https://github.com/robotology/icub-main/tree/master/app/skinGui/conf that serve to configure the GUI. Some of them do refer to files in the `positions` subdir.

However, address my request only if you have somehow tested your positions within the GUI as well.

**Comment by matejhof** [comment]
The file https://github.com/robotology/icub-main/blob/master/app/skinGui/conf/skinManAll.ini does not include the legs which, given the name of the file and the fact that iCub now comes with skin on the legs, is strange. Shall we consider adding the leg guis and position files to this? Or rename the file?

**Comment by pattacini** [comment]
Hi @matejhof 

>Shall we consider adding the leg guis and position files to this? Or rename the file?

Yep, we can extend the file to include the new positions.

Also, on second thought, I think it'd be valuable to add a README.md file within the `positions` directory with just a brief section telling how the new `.txt` files have been generated (i.e., with a pointer to your Gitlab project).

Better off making this information explicit rather than leaving it only in this PR.

**Comment by matejhof** [comment]
Ok, let's do that. Maybe we can do it with @rustlluk - test the new `skinManAll.ini` on the robot first, add the README, and then extend the PR with these changes or make a new PR. It will not happen in the next 2 or 3 weeks though.  In the meantime, shall we proceed with the current PR?

**Comment by pattacini** [comment]
Agree @matejhof 👍🏻 

I can add the README myself ex-post: it's very easy.
I'm going thus to merge this PR.

Thanks for the nice addition!

**Review by pattacini — APPROVED** [review]

**Pull request merged**